pkg/storage/azure/azure.go

```diff
@@ -1320,3 +1330,49 @@ func (d *driver) RemoveStorage(cr *imageregistryv1.Config) (retry bool, err erro
 func (d *driver) ID() string {
 	return d.Config.Container
 }
+
+// ensureUAMICredentials obtains and caches an Azure TokenCredential using a
+// User Assigned Managed Identity (UAMI).
+//
+// If MANAGED_AZURE_HCP_CREDENTIALS_FILE_PATH is unset, it returns (nil, false, nil).
+// When set, it loads a credential from a process-wide cache or creates one using the
+// provided Azure environment endpoints, stores it, and returns it.
+//
+// The bool result is true when a UAMI credential is available. An error is returned if
+// credential creation fails or a cached value has an unexpected type.
+//
+// ctx controls cancellation of credential creation. env supplies Azure endpoints.
+func (d *driver) ensureUAMICredentials(ctx context.Context, env autorestazure.Environment) (azcore.TokenCredential, bool, error) {
+	if stored, ok := globalAzureCredentials.Load(azureCredentialsKey); ok {
+		if cred, ok := stored.(azcore.TokenCredential); ok {
+			klog.V(2).Infof("Loaded UAMI credentials from cache")
+			return cred, true, nil
+		}
+		return nil, false, fmt.Errorf("expected cached credential to be azcore.TokenCredential")
+	}
+	if os.Getenv("MANAGED_AZURE_HCP_CREDENTIALS_FILE_PATH") == "" {
+		return nil, false, nil
```

Comment:
Is there a reason not to do this check first thing in the function? 🤔

Reply:
Yeah I don't think it will matter too much but in the principle of fail fast, I think it should be first.